Cli/Helper/ConfigurationHelper.php

```diff
@@ -90,32 +102,28 @@ private function askValue(InputInterface $input, OutputInterface $output, $varna
             $question = sprintf('What is your %s? ', $varname);
         }
 
-        $validator = function ($v) {
+        $validator = function ($v) use ($varname) {
             if (!$v) {
-                throw new \InvalidArgumentException('Your must answer this question.');
+                throw new \InvalidArgumentException(sprintf('Your must provide a %s!', $varname));
             }
 
             return $v;
         };
 
         $dialog = $this->getHelperSet()->get('dialog');
 
-        return $dialog->askAndValidate($output, $question, $validator, false, $default);
+        return $dialog->askAndValidate($output, $question, $validator, ($input->isInteractive() ? false : 1), $default);
```

Comment:
much better would be to avoid asking entirely in case the input is not interactive, by returning early in the method